src/SYNFPAN.m

```diff
@@ -0,0 +1,513 @@
+ . ;. s @eval@("labs",SYNZI,"status","issue")="VistA lab not found for loinc code: "_obscode_" "_labtype_" -- skipping"
+ . ;. s @eval@("status","errors")=$g(@eval@("status","errors"))+1
+ . s @eval@("labs",SYNZI,"parms","DHPLAB")=vistalab
+ . d log(jlog,"VistA Lab is: "_vistalab)
+ . s DHPLAB=vistalab
+ . ;
+ . s DHPOBS=value
+ . s recien=$o(^LAB(60,"B",DHPLAB,""))
+ . i recien="" d  ; oops lab test not found!!
+ . . S DHPLAB=$$UP^XLFSTR(DHPLAB)
+ . . s vistalab=DHPLAB
+ . . s recien=$o(^LAB(60,"B",DHPLAB,""))
+ . . d log(jlog,"VistA Lab is: "_vistalab)
+ . ;
+ . ; the following was made obsolete by changes Sam made to the Lab Data Import
+ . ;n xform s xform=$$GET1^DIQ(60,recien_",",410)
+ . ;n dec s dec=0
+ . ;i xform["S Q9=" d
+ . ;. s dec=+$p($p(xform,"""",2),",",3)
+ . ;;i $l($p(DHPOBS,".",2))>1 d
+ . ;i $l($p(DHPOBS,".",2))>0 d
+ . ;. s DHPOBS=$s(dec<4:$j(DHPOBS,1,dec),dec>3:$j(DHPOBS,1,3),1:$j(DHPOBS,1,0)) ; fix results with too many decimal places
+ . ; added for Covid tests
+ . i DHPOBS="" d  ; no quant value
+ . . n vtxt ; value text
+ . . s vtxt=$get(@json@("entry",SYNZI,"resource","valueCodeableConcept","text"))
+ . . ;i vtxt["Negative" s DHPOBS="Negative" ; 260385009
+ . . i vtxt["Negative" s DHPOBS="Not Detected" ; 260385009
+ . . ;i vtxt["Positive" s DHPOBS="Positive" ; 10828004
+ . . i vtxt["Positive" s DHPOBS="DETECTED" ; 10828004
+ . . i vtxt["Detected" s DHPOBS="DETECTED" ; 260373001
+ . . i vtxt["Not detected" s DHPOBS="Not Detected" ; 260415000
+ . . i vtxt["Confirmed" s DHPOBS="CONFIRMED" ; 
+ . s @eval@("labs",SYNZI,"parms","DHPOBS")=DHPOBS
+ . d log(jlog,"Value is: "_DHPOBS)
+ . ;
+ . i DHPLOINC="33914-3" d  q  ;
+ . . d log(jlog,"Skipping Estimated Glomerular Filtration Rate LOINC: 33914-3")
+ . ;
+ . s DHPUNT=unit
+ . s @eval@("labs",SYNZI,"parms","DHPUNT")=unit
+ . d log(jlog,"Units are: "_unit)
+ . ;
+ . s DHPDTM=hl7time
+ . s @eval@("labs",SYNZI,"parms","DHPDTM")=hl7time
+ . d log(jlog,"HL7 DateTime is: "_hl7time)
+ . ;
+ . s DHPPROV=$$MAP^SYNQLDM("OP","provider")
+ . n DHPPROVIEN s DHPPROVIEN=$o(^VA(200,"B",DHPPROV,""))
+ . if DHPPROVIEN="" S DHPPROVIEN=3
+ . s @eval@("labs",SYNZI,"parms","DHPPROV")=DHPPROVIEN
+ . d log(jlog,"Provider for outpatient is: #"_DHPPROVIEN_" "_DHPPROV)
+ . ;
+ . s DHPLOC=$$MAP^SYNQLDM("OP","location")
+ . n DHPLOCIEN s DHPLOCIEN=$o(^SC("B",DHPLOC,""))
+ . if DHPLOCIEN="" S DHPLOCIEN=4
+ . s @eval@("labs",SYNZI,"parms","DHPLOC")=DHPLOC
+ . d log(jlog,"Location for outpatient is: #"_DHPLOCIEN_" "_DHPLOC)
+ . ;
+ . s @eval@("labs",SYNZI,"status","loadstatus")="readyToLoad"
+ . ;
+ . ;i vistalab="PDW" q  ; skipping because it hangs - gpl wvehr 1/7/21
+ . ;i vistalab="SGPT" q  ; skipping it hangs loinc 1742-6 - gpl wvehr 1/7/21
+ . ;i vistalab="INFLUENZA A RNA" Q  ; likewise
+ . ;i vistalab="INFLUENZA B RNA" Q  ; likewise
+ . ;i vistalab="METAPNEUMOVIRUS RNA" Q  ; likewise
+ . ;
+ q 0
+ ;
+INITMAPS(LOC) ; initialize mapping table for panels
```

Comment:
In a future review, let's talk about supporting extra panels. There is one critical one that's missing now, but let's do that after the code fixes.

Reply:
ok